### PR TITLE
[self-hosted] Fix various issues around blobserve+ path based workspace access

### LINKED ISSUE #2570: [Self-Hosted] cannot start workspace: got 404 (GCP, nodomain)
### Describe the bug
When starting a workspace, got stuck at a blank page. Error when inspecting elements in Chrome: https://user-images.githubusercontent.com/24193764/102390481-38d34d80-3fb3-11eb-8839-04df40279ba8.png

Logs from the Dashboard component: 2020/12/16 18:29:08 [error] 19#19: *1152 open() "/www/data/dashboard/supervisor/frontend/main.js" failed (2: No such file or directory), client: 10.0.5.15, server: , request: "GET /supervisor/frontend/main.js HTTP/1.1", host: "35-233-172-32.ip.mygitpod.com", referrer: "https://35-233-172-32.ip.mygitpod.com/workspace/f08cc436-4ccc-47ea-a831-789ef862ae9c/" 10.0.5.15 - - [16/Dec/2020:18:29:08 +0000] "GET /supervisor/frontend/main.js HTTP/1.1" 404 555 "https://35-233-172-32.ip.mygitpod.com/workspace/f08cc436-4ccc-47ea-a831-789ef862ae9c/" "Mozilla/5.0 (X11; Linux x8664) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36" "10.138.0.5" 10.0.5.15 - - [16/Dec/2020:18:29:08 +0000] "GET /404/ HTTP/1.1" 200 667 "https://35-233-172-32.ip.mygitpod.com/workspace/f08cc436-4ccc-47ea-a831-789ef862ae9c/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36" "10.138.0.5"

### Steps to reproduce
Install following https://www.gitpod.io/docs/self-hosted/latest/install/install-on-gcp-script/, choosing `nodomain`

### PR DESCRIPTION
**Pull request by geropl**
Fixes #2570.

This PR contains various small fixes for re-enabling blobserve and path-based workspace access.

It's not complete, as we discussed stopping this effort and reduce functionality instead to relieve us from the maintenance burden. It's meant to be merge to a) document it, b) keep the minor fixes that are orthogonal on master nonetheless and c) untangle this PR from the duty to reduce the ingress variance as discussed.

# Test
it does not break preview env: http://gpl-test-sh-2.staging.gitpod-dev.com/workspaces/

### CODE REVIEW

File: components/ws-proxy/pkg/proxy/workspacerouter.go [hunk 1]
```diff
@@ -226,10 +227,15 @@ func getWorkspaceCoords(req *http.Request) WorkspaceCoords {
 // PathBasedTheiaRouter routes workspaces using a /workspaceID prefix.
 // Doesn't do port routing.
 func pathBasedTheiaRouter(r *mux.Router, wsInfoProvider WorkspaceInfoProvider, trimPrefix string) *mux.Router {
+	if trimPrefix == "" {
+		trimPrefix = "/"
+	}
+	if trimPrefix[len(trimPrefix)-1] != '/' {
+		trimPrefix = fmt.Sprintf("%s%s", trimPrefix, "/")
+	}
```
Comment (csweichel) [inline components/ws-proxy/pkg/proxy/workspacerouter.go:235]: nit

```suggestion
trimPrefix = strings.TrimSuffix(trimPrefix, "/") + "/"
```

File: components/ws-proxy/pkg/proxy/workspacerouter.go [hunk 2]
```diff
@@ -226,10 +227,15 @@ func getWorkspaceCoords(req *http.Request) WorkspaceCoords {
...
+
+	regex := regexp.MustCompile("^(" + trimPrefix + ")" + workspaceIDRegex)
```
Comment (csweichel) [inline components/ws-proxy/pkg/proxy/workspacerouter.go:237]: nit: what regex is this, i.e. what's its role (e.g. `prefixedWorkspaceIDRegex`)?

File: chart/values.yaml [hunk 3]
```diff
@@ -85,6 +83,26 @@ serviceWaiter:
...
+# - id: "Public-GitLab"
+#   host: "gitlab.com"
+#   type: "GitLab"
+#   oauth:
+#     clientId: "CLIENT_ID"
+#     clientSecret: "SECRET"
+#     callBackUrl: "https://gitpod.io/auth/gitlab/callback"
+#     settingsUrl: "https://gitlab.com/profile/applications"
+#   description: ""
+#   icon: ""
```
Comment (csweichel) [inline chart/values.yaml:105]: leftover?

If not, a comment would be nice to explain why this is there. E.g.
```YAML
# # Below are example authProvider configurations. Uncomment and complete values to use them.
# - id: ...
```

File: chart/values.yaml [hunk 4]
```diff
@@ -375,7 +393,7 @@ components:
       nodeRoots: 
       - /var/lib
       - /run/containerd/io.containerd.runtime.v1.linux/k8s.io
-      #- /run/containerd/io.containerd.runtime.v2.task/k8s.io
+      # - /run/containerd/io.containerd.runtime.v2.task/k8s.io
```
Comment (csweichel) [inline chart/values.yaml:396]: leftover?

**Review by csweichel — APPROVED** [review]
LGTM (a few nits)

File: chart/values.yaml (hunk 3 above)
Comment (geropl) [reply chart/values.yaml:106]: Added comment.

File: chart/values.yaml (hunk 4 above)
Comment (geropl) [reply chart/values.yaml:398]: Added comment.

**Pull request merged**